### PR TITLE
🌱 cron: allow controller to read CSVs from cloud storage (1/n)

### PR DESCRIPTION
**Pull request by spencerschrock**
#### What kind of change does this PR introduce?
Part of a series of refactors to generalize the cron infrastructure to enable re-use by the [ossf criticality score project](https://github.com/ossf/criticality_score)

- [X] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?
The cron job controller uses `cron/internal/data/projects.csv` as input for projects to analyze in the cron infrastructure.

#### What is the new behavior (if this is a feature change)?**
The controller will now read multiple CSV files from a configurable bucket if ran with no additional arguments. 
If passed one additional argument (which is how the controller is currently configured), the behavior is identical to the current behavior.
If passed multiple additional arguments, multiple local CSV files will be used to populate the pubsub topic.

- [X] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes
NONE

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
NONE
```

### CODE REVIEW

**Comment by spencerschrock** [comment]
@azeemshaikh38 @calebbrown 

This is the current state which handles importing multiple CSV files, either locally or from a single bucket (but not both).

Possible enhancements (could do both)
1. Structure the YAML in a way to define multiple buckets and pull CSVs from all of them
2. Pull from local files **AND** bucket files in the same run


What is the expected scale of project URLs? Currently the bucket implementation stores all files in the bucket in memory, which isn't a problem for the ~68MB scorecard `projects.csv` file, but could be a problem in the future.

File: cron/internal/data/iterator.go [hunk 1]
```diff
@@ -64,3 +64,30 @@ func (reader *csvIterator) Next() (RepoFormat, error) {
...
+		if i.iterators[i.current].HasNext() {
+			return true
+		}
+		i.current++
+	}
+	return false
+}
+
+func (i *nestedIterator) Next() (RepoFormat, error) {
+	r, err := i.iterators[i.current].Next()
```
Comment (azeemshaikh38) [inline cron/internal/data/iterator.go:88]: Replace with `return i.iterators[i.current].Next()` ?

**Review by azeemshaikh38 — APPROVED** [review]
Looks great!

File: cron/internal/data/iterator.go (hunk 1 above)
Comment (spencerschrock) [reply cron/internal/data/iterator.go:88]: I originally had that, but the `wrapcheck` linter complained.
> error returned from interface method should be wrapped: sig: func

**Comment by calebbrown** [comment]
Just a heads up, this adds support for CSV files that use the `data.RepoFormat` struct, and require a header row that matches.

Currently the Criticality Score enumerate_github tool outputs a TXT file with no header row - so I suspect this won't work for the current output as-is.

**Comment by oliverchang** [comment]
Would it make sense to support not having header rows, especially given that this is a machine readable format?

**Comment by azeemshaikh38** [comment]
> Would it make sense to support not having header rows, especially given that this is a machine readable format?

@spencerschrock fyi - https://github.com/jszwec/csvutil#but-my-csv-file-has-no-header-

**Comment by spencerschrock** [comment]
Will take a look and address in another PR

**Pull request merged**